### ISSUE TITLE
give Docker  Swarm a new life, involve the community and more developers in the development of Docker Swarm( Engine） 

### ISSUE THREAD

**Issue by DavidSche** [docker/roadmap#194]
Since 2018,Docker Swarm( Engine）development of new features has almost stopped ;the devops tools around swarm have also almost stopped updating in 2020, the most popular tools onDocker Swarm  , portainers development has also focused on turning to kubernetes in 2020.
 but  for some developers, Docker Compose +Docker  Engine(Swarm） remains the developer's preferred container solution.today,
Kubernetes has replaced docker engine with podman, and the update cycle of docker engine itself has slowed down. only compose or docker engine can continue to innovate and move forward; podman has started to support compose ;Docker Compose + kubenetes becoming the only solution?

 why not involve the community and more developers in the development of docker swarm? Let Docker Compose +Docker  Engine(Swarm） be the preferred option for developers and small companies to develop container clouds?

**Comment by BretFisher** [comment]
@DavidSche I agree with lots of your thoughts. The Compose Spec is great (even though it's limited vs. K8s manifests) and docker-compose is one of my favorite CLI tools.

Compose, the format, the CLI tool, and the standard around it have seen tons of progress in the last year. I've talked about a lot of that on my YouTube Live show, including the following episodes (all centered around the compose CLI and YAML format)
https://www.youtube.com/watch?v=MNyeTvCKhkM
https://www.youtube.com/watch?v=JPpyDfPyuJ0
and Docker has blogged about it:
https://www.docker.com/blog/compose-cli-aci-integration-now-available/
https://www.docker.com/blog/docker-compose-for-amazon-ecs-now-available/
And these new features are showing up in a new CLI that is included with Docker Desktop
https://github.com/docker/compose-cli

However, Swarm, the feature of the docker CLI and Engine, has received less attention, like you stated. In `20.10` engine version that was released in December 2020, we did get Jobs support, so check out `docker service create --help` to see the `--mode` new options for jobs.  At this point, Docker has no developers dedicated to Swarm issues or features, and it's largely up to Mirantis for developing new features and fixes.  IMO, Docker may still own the swarmkit project, but it's no longer something they are spending employee time on.

**Comment by DavidSche** [comment]
If you judge whether a project is viable from how many projects or DevOps tools are developed and collaborated around him, Docker Swarm will basically die after 2019, and the only active development project is Portainer.
Judging from the user's perspective, Docker Swarm is still alive because many developers and companies use it because it is simple to maintain and sufficient;

The practice of updating Docker Swarm once a year cannot attract more new users to participate. Let’s see how much Kubernetes is updated a year. Fast update is a feature of whether a project has vitality.

Mirantis has announced that the focus is not to invest in Docker Swarm; the reason for this is that Docker Enterprise customers are still using it, but they have been providing solutions to switch to Kubernetes, but the current program is that Kubernetesis still too complicated, so they proposed In order to build Docker Swarm over on Kubernetes , they have no plans for the future of Docker Swarm.
One reason Docker will regain its vitality in 2020 is because they focus on developers and involve the community, so why not let the community give Docker Swarm a better future?

**Comment by AkihiroSuda** [comment]
SwarmKit is being updated in this branch https://github.com/docker/swarmkit/commits/feature-volumes

**Comment by DavidSche** [comment]
If SwarmKit  can’t attract more developers and companies to participate, let more cloud native application developers pay attention to and participate in its development; let more open source projects and products support it, and use docker-compose to deploy on it ; Its destiny will definitely be death, whether it is fast or slow; maybe the same is true for docker engine, like openstack, no matter how brilliant it once was, it has lost focus and no more developers pay attention to it; maybe one day, The docker company will also become "Build, Ship Any App, Anywhere" , become a DevOps Tool Company that supports the kubernetes platform.

**Comment by decentral1se** [comment]
https://github.com/docker/compose-cli/issues/1350 :eyes:
